apps/devportal/data/markdown/pages/learn/accelerate/xm-cloud/implementation/redirect.md

```diff
@@ -0,0 +1,225 @@
+
+If you have a multiple redirects or you want to use regex you need to use redirect map instead. For example:
+
+```
+/company/products/arc-ow01 -> /products/oil/ow01
+/company/products/arc-ow02 -> /products/oil/ow02
+/company/products/arc-ow03 -> /products/oil/ow03
+```
+
+These type of redirects can be simplified with one regex statement in redirect map item:
+
+```
+/company/products/arc-(.*)$  -> /products/oil/$1
+```
+
+  <Alert status="info">
+    <AlertIcon />
+      IMPORTANT: Always keep in mind the redirects are processed with every single request, if your regex is too complicated, it my cause a performance issue.
+  </Alert>
+
+### How do SXA redirects work?
+
+With every request, the redirect middleware in your Next.js application will run the following graphql query against the experience edge to get the list of redirects:
+
+```graphql
+query RedirectsQuery($siteName: String!) {
+  site {
+    siteInfo(site: $siteName) {
+      redirects {
+        pattern
+        target
+        redirectType
+        isQueryStringPreserved
+        locale
+      }
+    }
+  }
+}
+```
+
+After receiving the redirects list, the jss redirect middleware will process it against the current request.
+
+### Redirects are not updated
+
+After publishing the redirect map item, you need to publish the headless site item (your site as a single item). If you forget this step, the query above will return the outdated results.
+
+### Considerations for SXA redirects:
+
+If you are planning to use the redirect item or redirect map consider the following:
+
+- Whereever possible try to use Regex instead of creating multiple redirect entry.
+- Keep the number of SXA redirect as low as possible, as mentioned before the redirect middleware needs to process the list by hitting experience edge endpoint and it can cause performance issue.
+- Periodically check the number of redirects in the XM Cloud and if needed move them to the frontend app (see the next part).
+
+### Middleware based redirects for Next.js
+
+When your redirects count exceeds 1024 redirects you need to use a middleware. In the related documentation section you can find provided example using edge function.
+If you do not want to use edge config, you can still use a edge middleware with a simple json file that includes the redirect list.
+
+This is an simplified example for `/site-redirects.json`:
+
+```json
+{
+  "hostName": ["www.mysite.com", "mysite.com"],
+  "redirects": [
+    {
+      "s": "/old-page-url-1",
+      "d": "/en/new-page-url-1",
+      "c": 301
+    },
+    {
+      "s": "/old-page-url-2",
+      "d": "https://external-site.com",
+      "c": 302
+    }
+  ]
+}
+```
+
+And inside your `/src/middleware.ts`:
```

Comment:
Can we re-write this code example to use the JSS pattern of middleware plugins please?